drivers/usbdev_synopsys_dwc2/usbdev_synopsys_dwc2.c

```diff
@@ -674,6 +705,41 @@ static void _usbdev_init(usbdev_t *dev)
     if (usb_new_phy(&phy_conf, &phy_hdl) != ESP_OK) {
         LOG_ERROR("usbdev: Install USB PHY failed\n");
     }
+
+#elif defined(MCU_EFM32)
+
+    /* Block EM2 and EM3. In EM2, most USB core registers are reset and the
+     * FIFO content is lost. EM2 is unblocked during USB suspend */
+    pm_block(EFM32_PM_MODE_EM3);
+    pm_block(EFM32_PM_MODE_EM2);
+
+#if defined(CPU_FAM_EFM32GG12B)
+    /* enable USHFRCO for USBCLK and set the tuning to 48 MHz */
+    CMU_OscillatorEnable(cmuOsc_USHFRCO, true, true);
+    CMU_USHFRCOBandSet(cmuUSHFRCOFreq_48M0Hz);
+    /* enable USB clock recovery */
+    CMU->USBCRCTRL = CMU_USBCRCTRL_USBCREN;
+    /* select USHFRCO as USB PHY clock source */
+    CMU->USBCTRL = CMU_USBCTRL_USBCLKSEL_USHFRCO | CMU_USBCTRL_USBCLKEN;
+#elif defined(CPU_FAM_EFM32GG) || defined(CPU_FAM_EFM32LG)
+    /* enable USB system clock */
+    CMU->HFCORECLKEN0 |= CMU_HFCORECLKEN0_USB;
+    /* select HFCLK as USB PHY clock source */
+    CMU->CMD |= CMU_CMD_USBCCLKSEL_HFCLKNODIV;
+    /* enable USB core clock */
+    CMU->HFCORECLKEN0 |= CMU_HFCORECLKEN0_USBC;
+#else
+#error "EFM32 family not yet supported"
+#endif
```

Comment:
Maybe, `CMU_ClockSelectSet()` could be used here, too.

Reply:
Hm, I already thought about that. The code here is from the reference manuals and `CMU_ClockSelectSet` doesn't cover all required settings. It enables the oscillator if necessary and sets `CMU_USBCTRL_USBCLKSEL_USHFRCO` respectively `CMU_CMD_USBCCLKSEL_HFCLKNODIV` in the according register.

Reply:
Unfortunately, I don't have a board based on these CPUs at hand.

Reply:
Hm, who could have such a board at hand?

Reply:
Maybe @basilfx has? He introduced this families into RIOT.

Reply:
Hm, @basilfx didn't react :worried: I hope @chrysn will have some time to test it, it seems that he has a `stk3700` board.

Reply:
Code-wise the efm32-part looks good to me!